### DISCUSSION TITLE
威联通上部署分享

### DISCUSSION THREAD

**Discussion by Qiqia** [oddfar/campus-imaotai#67] [分享部署教程]
威联通上部署成功，网页可以打开，i茅台账号配置了，但是能不能申购成功还有待测试，分享下注意点吧
1.java8要换成openjdk8，第一次build拉镜像在国内花了很久。
2.nginx的conf文件里面增加一条user root，以root启动nginx，至于为什么是root，暂时不太清楚。
3.compose.yml文件里面，mysql的配置里面增加 user： xxx:xxx 这里面对应uid和gid，不能用0:0，具体原因不太清楚，因为用0:0就报错没有权限。
4.compoes.yml文件里面，nginx的配置里面增加 user: 0:0 就是root启动，与第二点呼应。
5.在添加i茅台账号的时候，发短信失败，issus里面多说是网的问题，可以在点发送验证码的时候，同时在手机上发送一个验证码。然后用那个验证码登录，暂时不知道有什么隐患。
6.刚配置好，申购成功了，可以自动去旅行。

**Comment by bmybmy8** [comment]
大佬，mysql的配置里面增加 user： xxx:xxx 这里面对应uid和gid。请问这个uid和gid如何获取啊。谢谢啦。

**Reply by Qiqia** [reply]
1、通过查看“/etc/passwd”文件来确定自己的uid和gid，查看命令如“cat /etc/passwd”；2、直接通过id命令查看当前用户的id即可。

**Comment by bmybmy8** [comment]
> 1、通过查看“/etc/passwd”文件来确定自己的uid和gid，查看命令如“cat /etc/passwd”；2、直接通过id命令查看当前用户的id即可。

谢谢大佬，我再试试看。

**Comment by vip0917** [comment]
按照大佬说的，加入了user，成功安装了，但是容器一启动很快就意外停止

**Comment by TangTree** [comment]
大佬，nginx的conf文件是哪一个

**Reply by Qiqia** [reply]
你不是在这个项目下载了一个文件么，文件解压后进入到nginx那个文件夹里面 会有一个conf文件

**Reply by dingchenghu0206** [reply]
大佬，conf文件里面再哪里添加user
